lib/modules/userTagger.js

```diff
@@ -940,25 +926,68 @@ modules['userTagger'] = {
-		var thisComment = thisAuthorObj.parentNode.parentNode.querySelector('.usertext');
-		if (thisComment) {
-			var showLink = document.createElement('a');
-			showLink.textContent = 'show anyway?';
-			showLink.setAttribute('href','#');
-			showLink.addEventListener('click', function(e) {
-				$(this).parent().html($(this).parent().attr('data-original')).removeClass('ignoredUserComment');
+	regexes: {
+		front: /^https?:\/\/(?:[\-\w\.]+\.)?reddit\.com\/?$/i
+	},
+	ignorePost: function(thing) {
+		var title;
+		if (modules['userTagger'].options.hardIgnore.value) {
+			thing['element'].parentNode.removeChild(thing['element']);
+		} else {
+			title = thing.entry.querySelector('p.title');
+			if (title === null) {
+				title = thing.querySelector('.search-result-header');
+				$(thing['element']).find('.search-result-body > .md, .search-expando-button, .search-result-footer').hide();
+			}
+			title.setAttribute('data-original', title.innerHTML);
+			title.innerHTML = '';
+			title.classList.remove('title');
+			$(thing['element']).find('.expando-button, .usertext-body, a.thumbnail').hide();
+			title.appendChild(modules['userTagger'].ignoredPlaceholder(thing, title));
+		}
+	},
+	ignoreComment: function(thing) {
+		var body, toggleComment;
+		if (modules['userTagger'].options.hardIgnore.value) {
+			if (!$(thing['element']).children('.child').children().length) {
+				thing['element'].parentNode.removeChild(thing['element']);
+			}
+			else if (thing['element'].classList.contains('noncollapsed')) {
+				toggleComment = thing.entry.querySelector('a.expand');
+				RESUtils.click(toggleComment);
+			}
+		}
+		body = thing.entry.querySelector('.md-container > .md');
+		body.setAttribute('data-original', body.innerHTML);
+		body.innerHTML = '';
+		$(thing['element']).find('.buttons').hide();
+		body.appendChild(modules['userTagger'].ignoredPlaceholder(thing, body));
+	},
+	// Inline message to indicate when content has been softly hidden.
+	ignoredPlaceholder: function(thing, target) {
+		var wrapper = RESUtils.createElement('span', null, 'res-ignored-message');
+		var icon = RESUtils.createElement('span', null, 'res-icon', '\uF093');
+		var text = RESUtils.createElement('span', null, null, ' ignored. ');
+		var button = RESUtils.createElement('a', null, null, 'Show anyway.');
+		button.setAttribute('href', '#');
+		wrapper.appendChild(icon);
+		wrapper.appendChild(text);
+		// Click to unhide ignored content.
+		if (modules['userTagger'].options.showIgnored.value) {
+			wrapper.appendChild(button);
+			button.addEventListener('click', function(e) {
 				e.preventDefault();
-			});
-			// store original
-			thisComment.setAttribute('data-original', thisComment.innerHTML);
-			// remove comment
-			thisComment.textContent = thisAuthor + ' is an ignored user.';
-			thisComment.appendChild(showLink);
-			thisComment.classList.add('ignoredUserComment');
-			return thisComment;
+				target.innerHTML = target.getAttribute('data-original');
```

Comment:
Turns out safeHtml() strips out useful classes from the content, like imgScanned, loggedin, etc, affecting styles.